### DISCUSSION TITLE
fatal error: chardet.h: No such file or directory

### DISCUSSION THREAD

**Discussion by Zwei1996** [linuxdeepin/developer-center#3291] [Q&A | 问答板块]
when i was in the make procedure, i got this issue fatal error: chardet.h: No such file or directory，could you please tell me how to solve it? 我查了一下，这个貌似是个python模块，网上的c头文件几乎没有，有的的话，make出来就是dpdfglobal.cpp:42:5: error: ‘DetectObj’ was not declared in this scope
   42 |     DetectObj *obj = detect_obj_init();
，注释掉问题代码又会提示我缺少chardet的动态链接库，请问如何解决

**Comment by Zwei1996** [comment]
补充一下，是deepin-reader的安装

**Reply by BLumia** [reply]
是在尝试在哪个发行版下构建？描述下你已经操作过的步骤？（ps. 你可以直接点右上角三个点，选 Edit 来编辑原主题的内容）

**Reply by Zwei1996** [reply]
下载的是master分支的源码，步骤就是先装依赖，readme提到的环境都没有问题，然后在make这里，会报缺失一些头文件，前面的都解决了，最后是chardet.h这个头文件我找不到c版本是哪个库涉及的，我试了一下https://github.com/sspring/chardet这个上面的头文件，但是会报dpdfglobal.cpp:42:5: error: ‘DetectObj’ was not declared in this scope这个错误，应该是缺啥动态链接库

**Reply by BLumia** [reply]
哪个发行版？如果是 deepin 的话，可以用 `apt build-dep deepin-reader` 安装所有构建依赖（有时候 README 会更新不及时）

**Reply by Zwei1996** [reply]
是deepin的，https://github.com/linuxdeepin/deepin-reader，不好意思没表达清楚，谢谢您，我回去再试试

**Reply by Zwei1996** [reply]
> 哪个发行版？如果是 deepin 的话，可以用 `apt build-dep deepin-reader` 安装所有构建依赖（有时候 README 会更新不及时）

您好，我ubuntu20下面提示这个错误，E: Unable to find a source package for deepin-reader，我用的是阿里的源，请问还需要在source.list下面添加别的源吗

**Reply by BLumia** [reply]
旧版本的话需要你手动安装相应依赖了，不过如果没有别的原因的话，还是强烈建议升级到22.04的（22.04 是长期支持版本）。依赖列表在源码根目录下的 debian 目录里的 control 文件里有列出，也可以参照那个列表手动安装。
